applications/GeoMechanicsApplication/python_scripts/element_test/ui_builder.py

```diff
@@ -0,0 +1,293 @@
+
+    def _run_simulation(self):
+        clear_log()
+        try:
+            log_message("Starting calculation... Please wait...", "info")
+            log_message("Validating input...", "info")
+            self.root.update_idletasks()
+
+            umat_params = [e.get() for e in self.entry_widgets.values()]
+
+            if self.current_test.get() != "Triaxial":
+                raise NotImplementedError(f"{self.current_test.get()} simulation not yet implemented.")
+
+            eps_max = float(self.triaxial_widgets[MAX_STRAIN_LABEL].get())
+            sigma_init = float(self.triaxial_widgets[INIT_PRESSURE_LABEL].get())
+            n_steps = float(self.triaxial_widgets[NUM_STEPS_LABEL].get())
+            duration = float(self.triaxial_widgets[DURATION_LABEL].get())
+
+            if any(val <= 0 for val in [eps_max, n_steps, duration]) or sigma_init < 0:
+                raise ValueError("All values must be positive and non-zero.")
+
+            cohesion_phi_indices = None
+            if self.mohr_checkbox.get():
+                cohesion_phi_indices = (int(self.cohesion_var.get()), int(self.phi_var.get()))
+
+            index = self.model_dict["model_name"].index(self.model_var.get()) + 1 if self.dll_path else -2
+
+            log_message("Calculating...", "info")
+            self.root.update_idletasks()
+
+            figs = run_triaxial_simulation(
+                dll_path=self.dll_path or "",
+                index=index,
+                umat_parameters=[float(x) for x in umat_params],
+                num_steps=n_steps,
+                end_time=duration,
+                maximum_strain=eps_max,
+                initial_effective_cell_pressure=sigma_init,
+                cohesion_phi_indices=cohesion_phi_indices
+            )
+            self.root.after(0, lambda: render_plots(figs, self.axes, self.canvas))
+            log_message("Simulation completed successfully.", "info")
+
+        except Exception:
+            log_message("An error occurred during simulation:", "error")
+            log_message(traceback.format_exc(), "error")
+
+        finally:
+            self.root.after(0, self._unfreeze_gui)
+            self.is_running = False
+
+    def _enable_run_button(self):
+            self.run_button.config(state="normal")
+            self.is_running = False
+
+    def _set_widget_state(self, parent, state):
+        for child in parent.winfo_children():
+            if isinstance(child, (ttk.Entry, ttk.Combobox, tk.Button, ttk.Button, tk.Checkbutton, ttk.Checkbutton)):
+                child.configure(state=state)
+            elif isinstance(child, scrolledtext.ScrolledText):
+                child.config(state=state if state == "normal" else "disabled")
+            elif isinstance(child, (ttk.Frame, tk.Frame)):
+                self._set_widget_state(child, state)
+
+        for widget in self.external_widgets:
+            widget.configure(state=state)
+
+    def _freeze_gui(self):
```

Comment:
I'd prefer disable/enable for these functions, sounds a bit nicer than freeze/unfreeze
```suggestion
    def _disable_gui(self):
```